### PR TITLE
freethreading support

### PR DESCRIPTION
**Pull request by isuruf**
Checklist
* [ ] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [ ] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -39,7 +39,8 @@ source:
   # https://github.com/numpy/numpy/tree/v{{ version }}/vendored-meson
   - folder: vendored-meson/meson
     git_url: https://github.com/numpy/meson.git
-    git_rev: 6f88e485f27bb0a41d31638f0c55055362e0b1ac
+    git_rev: 6f88e485f27bb0a41d31638f0c55055362e0b1ac  # [not (win and is_freethreading)]
+    git_rev: 5f7159556a40844cba411e3a45d61192e36bf9ad  # [win and is_freethreading]
```
Comment (h-vetinari) [inline recipe/meta.yaml:43]: Where did you get that commit from? I don't find it in https://github.com/numpy/meson

Comment (ocefpaf) [reply recipe/meta.yaml:43]: To find commits you need to paste them after the `/commit/` part of the URL. Here it is:

https://github.com/numpy/meson/commit/5f7159556a40844cba411e3a45d61192e36bf9ad

Comment (h-vetinari) [reply recipe/meta.yaml:43]: Thanks. I do know that part, but what I meant by "I don't find it" is the yellow box here

image

It's not an obvious branch/tag, nor does the submodule in numpy point to that commit, so I'm curious where this comes from. Probably some PR somewhere?

Comment (isuruf) [reply recipe/meta.yaml:43]: Came from https://github.com/numpy/meson/pull/15

Comment (ocefpaf) [reply recipe/meta.yaml:43]: The message is just that. The commit is not on that repository yet. As this is all experimental and new I would expecte it to be in a PR, as Isuru just posted above.

Comment (h-vetinari) [reply recipe/meta.yaml:43]: > As this is all experimental and new I would expecte it to be in a PR, as Isuru just posted above.

Which is what I was interested in. ;)

I'd leave a comment, but not a blocking concern.

**Review by h-vetinari — APPROVED** [review]
Good from my side

**Pull request merged**